lib/api/src/js/imports.rs

```diff
@@ -233,29 +238,40 @@ impl Imports {
 }
 
 impl AsJs for Imports {
+    #[allow(unused_unsafe)]
     fn as_jsvalue(&self, store: &impl AsStoreRef) -> wasm_bindgen::JsValue {
         let imports_object = js_sys::Object::new();
         for (namespace, name, extern_) in self.iter() {
-            let val = js_sys::Reflect::get(&imports_object, &namespace.into()).unwrap();
+            let val = unsafe { js_sys::Reflect::get(&imports_object, &namespace.into()).unwrap() };
             if !val.is_undefined() {
                 // If the namespace is already set
-                js_sys::Reflect::set(
-                    &val,
-                    &name.into(),
-                    &extern_.as_jsvalue(&store.as_store_ref()),
-                )
-                .unwrap();
+                #[allow(unused_unsafe)]
+                unsafe {
+                    js_sys::Reflect::set(
+                        &val,
+                        &name.into(),
+                        &extern_.as_jsvalue(&store.as_store_ref()),
+                    )
+                    .unwrap();
+                }
             } else {
                 // If the namespace doesn't exist
                 let import_namespace = js_sys::Object::new();
-                js_sys::Reflect::set(
-                    &import_namespace,
-                    &name.into(),
-                    &extern_.as_jsvalue(&store.as_store_ref()),
-                )
-                .unwrap();
-                js_sys::Reflect::set(&imports_object, &namespace.into(), &import_namespace.into())
+                #[allow(unused_unsafe)]
+                unsafe {
```

Comment:
Why we are doing this? `#[allow(unused_unsafe)]` and then `unsafe {`. Doesn't make that much sense.
Perhaps I'm missing something?

Reply:
Discussed in sibling threads.

Code now has an explanation comment.